lib/pqa/import.rb

```diff
@@ -7,21 +7,31 @@ def initialize(pqa_service = nil)
     end
 
     def run(date_from, date_to)
-      init_state!
+      query_api_and_update(type: :date_range, date_from: date_from, date_to: date_to)
+    end
 
-      ActiveRecord::Base.transaction do
-        questions = @pqa_service.questions(date_from, date_to)
-        @total    = questions.size
+    def run_for_question(uin)
+      query_api_and_update(type: :uin, uin: uin)
+    end
+
+
+    private
 
-        questions.each do |q|
-          insert_or_update(q)
+    def query_api_and_update(options)
+      init_state!
+      ActiveRecord::Base.transaction do
+        questions = []
+        if options[:type] == :date_range
```

Comment:
Why using a block here instead of switching on `options[:type]`?

```ruby
def run(date_from, date_to)
   insert_and_update { @pqa_service.questions(date_from, date_to) }
end

def run_for_question(uin)
   insert_or_update { @pqa_service.question(uin) }
end

```